### ISSUE TITLE
Launch DOS Prompt will frozen whole Windows 95 if Set machine  = svga_s3trio64v+ in config 

### ISSUE THREAD

**Issue by wildtang** [joncampbell123/dosbox-x#5333]
### Describe the bug

Launch DOS Prompt will frozen whole Windows 95 if Set machine  = svga_s3trio64v+ in config.
Using ver.2024.12.04 (VS SDL2 32-bit) Build date/time: Dec 05, 2024 01:00:19am

### Steps to reproduce the behaviour

I have a Windows 95 guest(4.00.950a).

If I set machine to svga_s3trio64v+, startup Windows 95, install video driver (use Windows 95/98 Driver v4.10.01.2108-2.41.07 on https://www.dosdays.co.uk/topics/Manufacturers/s3_downloads.php#Trio64VPlus, zip link is https://www.dosdays.co.uk/media/s3/Trio64V+/w9524107.zip), then restart Win 95, click MS-DOS Prompt on the Desktop, the whole Win 95 becomes frozon, I have to restart Win 95 forcely.

But, If I revert machine to svga_s3 in config, and change video drvier to Adapter Type to S3 Trio32/64 PCI, I can launch MS-DOS Prompt as expect.

### What operating system(s) this bug have occurred on?

Windows 10 LTSC 21H2 (19044.5131)

### What version(s) of DOSBox-X have this bug?

ver.2024.12.04 (VS SDL2 32-bit) Build date/time: Dec 05, 2024 01:00:19am

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by joncampbell123** [comment]
Huh. On my setup, Windows 95 recognizes that machine type, but won't let me set anything other than the stock 640x480 16-color mode.

**Comment by joncampbell123** [comment]
Windows 98 does recognize it but RGB colors in 32bpp accelerated graphics mode are broken.

**Comment by joncampbell123** [comment]
Ah, and launching the DOS prompt in Windows 98 hangs as described.

**Comment by maxpat78** [comment]
@wildtang try to change CPU type with menu, that should defrost your emulation and make the command prompt window appear.

**Comment by wildtang** [comment]
> @wildtang try to change CPU type with menu, that should defrost your emulation and make the command prompt window appear.
> 
> See #4265 also.

Yeah, As my tests today, If I click menu CPU→CPU Core, changes it from Dynamic core(dynamic_x86) to Normal core, DOS prompt window appeares.

Change menu CPU→CPU type has no effect to this issue.

@joncampbell123 @maxpat78 
Hope the info I provided here can help identify and resolve this problem, good luck to you.

**Comment by joncampbell123** [comment]
> > @wildtang try to change CPU type with menu, that should defrost your emulation and make the command prompt window appear.
> > See #4265 also.
> 
> Yeah, As my tests today, If I click menu CPU→CPU Core, changes it from Dynamic core(dynamic_x86) to Normal core, DOS prompt window appeares.
> 
> Change menu CPU→CPU type has no effect to this issue.
> 
> @joncampbell123 @maxpat78 Hope the info I provided here can help identify and resolve this problem, good luck to you.

Ah, wait a minute.
That reminds me of a bug I found with Windows 95 OSR2 where, if you start a DOS prompt and DOSBox-X is not doing any segment limit checks (against the GDT), Windows hangs. As far as I was able to determine, the DOS VM required a segment limit violation exception to start the DOS environment.

Dynamic core, for obvious performance reasons, doesn't do any segment limit checks.

**Comment by joncampbell123** [comment]
Weird things like that are fairly normal for Microsoft to do. Consider for example that NTVDM.EXE in Windows NT/2000/XP allows DOS code within the VM to access a very simple API using a specific invalid opcode pattern. DOSLIB has a proof of concept. Basically you can use it to load and unload VDM DLLs into NTVDM.EXE.
